submarine-server/server-core/src/test/java/org/apache/submarine/server/rest/ExperimentRestApiTest.java

```diff
@@ -0,0 +1,182 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.submarine.server.rest;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonArray;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.submarine.server.SubmarineServer;
+import org.apache.submarine.server.api.experiment.Experiment;
+import org.apache.submarine.server.api.experiment.ExperimentId;
+import org.apache.submarine.server.api.experiment.ExperimentLog;
+import org.apache.submarine.server.api.spec.EnvironmentSpec;
+import org.apache.submarine.server.api.spec.ExperimentSpec;
+import org.apache.submarine.server.experiment.ExperimentManager;
+import org.apache.submarine.server.gson.ExperimentIdDeserializer;
+import org.apache.submarine.server.gson.ExperimentIdSerializer;
+import org.junit.Test;
+import org.junit.BeforeClass;
+import org.junit.Before;
+import org.junit.After;
+
+
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ExperimentRestApiTest {
+
+  private static ExperimentRestApi experimentRestApi;
+  private static ExperimentManager mockExperimentManager;
+  private final AtomicInteger experimentCounter = new AtomicInteger(0);
+  private Experiment experiment;
+
+  private static GsonBuilder gsonBuilder = new GsonBuilder()
+      .registerTypeAdapter(ExperimentId.class, new ExperimentIdSerializer())
+      .registerTypeAdapter(ExperimentId.class, new ExperimentIdDeserializer());
+  private static Gson gson = gsonBuilder.setDateFormat("yyyy-MM-dd HH:mm:ss").create();
+
+  @BeforeClass
+  public static void init() {
+    mockExperimentManager = mock(ExperimentManager.class);
+    experimentRestApi = new ExperimentRestApi();
+    experimentRestApi.setExperimentManager(mockExperimentManager);
+  }
+
+  @Before
+  public void createAndUpdateExperiment() {
+    experiment = new Experiment();
+    experiment.setAcceptedTime("2020-08-06T08:39:22.000+08:00");
+    experiment.setCreatedTime("2020-08-06T08:39:22.000+08:00");
+    experiment.setRunningTime("2020-08-06T08:39:23.000+08:00");
+    experiment.setFinishedTime("2020-08-06T08:41:07.000+08:00");
+    experiment.setUid("0b617cea-81fa-40b6-bbff-da3e400d2be4");
+    experiment.setName("tf-example");
+    experiment.setStatus("Succeeded");
+    experiment.setExperimentId(ExperimentId.newInstance(SubmarineServer.getServerTimeStamp(),
+        experimentCounter.incrementAndGet()));
+    ExperimentSpec experimentSpec = new ExperimentSpec();
+    EnvironmentSpec environmentSpec = new EnvironmentSpec();
+    environmentSpec.setName("foo");
```

Comment:
There are some other properties (`dockerImage`, `kernelSpec`, ..etc )in EnvironmentSpec, we need to set them as well. 
https://github.com/apache/submarine/blob/master/submarine-server/server-api/src/main/java/org/apache/submarine/server/api/spec/EnvironmentSpec.java

We should also set meta for `experimentSpec`
https://github.com/apache/submarine/blob/master/submarine-server/server-api/src/main/java/org/apache/submarine/server/api/spec/ExperimentSpec.java

btw, I didn't see `updateExperiment` in this `createAndUpdateExperiment` function